pilot/helpers/cli.py

```diff
@@ -84,32 +85,50 @@ def terminate_running_processes():
 
 
 def term_proc_windows(pid: int):
-   try:
-       subprocess.run(["taskkill", "/F", "/T", "/PID", str(pid)])
-   except subprocess.CalledProcessError as e:
-       logger.error(f'Error while terminating process: {e}')
+    try:
+        subprocess.run(["taskkill", "/F", "/T", "/PID", str(pid)])
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Error while terminating process: {e}')
 
 
 def term_proc_unix_like(pid: int):
-   try:
-       os.killpg(pid, signal.SIGKILL)
-   except OSError as e:
-       logger.error(f'Error while terminating process: {e}')
+    try:
+        os.killpg(pid, signal.SIGKILL)
+    except OSError as e:
+        logger.error(f'Error while terminating process: {e}')
 
 
-def terminate_process(pid: int, name=None) -> None:
-   if name is None:
-       name = ''
-   logger.info('Terminating process "%s" (pid: %s)', name, pid)
+def is_process_running(pid: int) -> bool:
+    """Check if there is a running process with the given PID."""
+    try:
+        # psutil.NoSuchProcess will be raised if the process doesn't exist
+        process = psutil.Process(pid)
+        return process.is_running()
+    except psutil.NoSuchProcess:
+        return False
 
-   if platform.system() == "Windows":
-       term_proc_windows(pid)
-   else:  # Unix-like systems
-       term_proc_unix_like(pid)
 
-   for command_id in list(running_processes.keys()):
-       if running_processes[command_id][1] == pid:
-           del running_processes[command_id]
+def terminate_process(pid: int, name=None) -> None:
```

Comment:
If we're using `psutil`, we can use its cross-platform utilities for terminating the processes, such as `psutil.Process().kill()`.

They even have a recipe on how to terminate a process (but itself) and all of its children here: [kill_process_tree()](https://psutil.readthedocs.io/en/latest/#kill-process-tree). I have adapted it to our case as a suggestion (note I haven't actually tested it):

```python
def kill_proc_tree(pid):
    """Kill a process tree (including grandchildren)."""

    pid = int(pid)
    if pid == os.getpid():
        # Log this because this is a bug in the program
        log.error(f"Asked to kill myself, I won't")
        return

    try:
        parent = psutil.Process(pid)
        if not parent.is_running():
            raise psutil.NoSuchProcess()
    except psutil.NoSuchProcess:
        log.warning(f"No such process: {pid}")
        return

    children = parent.children(recursive=True)
    to_kill = children + [parent]

    for p in to_kill:
        try:
            p.kill()
        except psutil.NoSuchProcess:
            # we don't care if the process is no longer there
            pass
        except Exception as e:
            # we care about all the other possible errors
            log.warning(f"Error killing process {pid}: {e}", exc_info=True)

    # Wait at most 1s for everything to terminate, so we
    # can reap the children instead of making them zombies.
    psutil.wait_procs(to_kill, timeout=1)
```